go/nbs/table_persister.go

```diff
@@ -184,13 +187,13 @@ func (ftp fsTablePersister) Compact(mt *memTable, haver chunkReader) (name addr,
 		io.Copy(temp, bytes.NewReader(data))
 		return temp.Name(), name, chunkCount
 	}()
-	if chunkCount > 0 {
-		err := os.Rename(tempName, filepath.Join(ftp.dir, name.String()))
-		d.PanicIfError(err)
-	} else {
+	if chunkCount == 0 {
 		os.Remove(tempName)
+		return emptyChunkSource{}
 	}
-	return name, chunkCount
+	err := os.Rename(tempName, filepath.Join(ftp.dir, name.String()))
+	d.PanicIfError(err)
+	return newMmapTableReader(ftp.dir, name, chunkCount)
```

Comment:
I considered that and decided that it would ugly up the API for not enough gain. These bytes were just written, so I expect them to be in the OS filesystem cache, meaning that it'll be pretty cheap to read them again.